xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-api/src/main/java/org/xwiki/notifications/page/PageNotificationEventUpdater.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.notifications.page;
+
+import org.xwiki.component.annotation.Role;
+import org.xwiki.notifications.internal.page.PageNotificationEventDescriptorManager;
+import org.xwiki.stability.Unstable;
+
+/**
+ * This role allows the {@link PageNotificationEventDescriptorManager} to always have
+ * a up-to-date list of {@link PageNotificationEventDescriptor} to work on.
+ *
+ * @version $Id$
+ * @since 9.5RC1
+ */
+@Role
+@Unstable
+public interface PageNotificationEventUpdater
+{
+    /**
+     * Fetch every registered PageNotificationEventDescriptorClass XObjects in the wiki, then update the
+     * {@link PageNotificationEventDescriptorManager}.
+     */
+    void updateDescriptors();
```

Comment:
I thought that separating the updater from the PageNotificationEventDescriptorManager would be a good idea because it would allow to better customize the way those descriptors are updated.